catalog/aws.go

```diff
@@ -281,9 +271,17 @@ func (a *awsSyncer) fetchNodes(id string) ([]awssdtypes.InstanceSummary, error)
 }
 
 func (a *awsSyncer) discoverNodes(name string) ([]awssdtypes.InstanceSummary, error) {
+	if a.namespace.Properties == nil ||
+		a.namespace.Properties.HttpProperties == nil ||
+		a.namespace.Properties.HttpProperties.HttpName == nil {
+		return nil, fmt.Errorf("namespace properties are nil")
+	}
+
 	resp, err := a.client.DiscoverInstances(context.TODO(), &awssd.DiscoverInstancesInput{
-		HealthStatus:  awssdtypes.HealthStatusFilterHealthy,
-		NamespaceName: aws.String(a.namespace.name),
+		HealthStatus: awssdtypes.HealthStatusFilterHealthy,
+		// This is the http name, which can be different from the display name in the event there have been
+		// multiple versions of the namespace (i.e. it has been recreated).
+		NamespaceName: a.namespace.Properties.HttpProperties.HttpName,
```

Comment:
BUG: The key difference is using the `HttpName` instead of the display name. I found this bug by accident by recreating the same namespace name multiple times. The `HttpName` is unique each time.